csharp/ql/lib/semmle/code/csharp/security/dataflow/ZipSlipQuery.qll

```diff
@@ -3,147 +3,360 @@
+    this.getReturnType() instanceof BoolType and
+    this.getAParameter() = paramFilename
+  }
 
-  deprecated override predicate isSanitizerGuard(DataFlow::BarrierGuard guard) {
-    guard instanceof SanitizerGuard
+  Parameter paramFilePath() { result = paramFilename }
+}
+
+class DirectWrapperSantizierMethod extends AbstractWrapperSanitizerMethod {
+  /*
+   * A DirectWrapperSantizierMethod is a Method where
+   *      The function can /only/ returns true when passes through the RootSanitizerGuard
+   *
+   *     bool wrapperFn(a,b){
+   *       if(guard(a,b))
+   *         return true
+   *       ....
+   *       return false
+   *     }
+   *
+   *     bool wrapperFn(a,b){
+   *       ...
+   *       return guard(a,b)
+   *     }
+   */
+
+  DirectWrapperSantizierMethod() {
+    forex(ReturnStmt ret | ret.getEnclosingCallable() = this |
+      ret.getExpr().(BoolLiteral).getBoolValue() = false
+      or
+      exists(ZipSlipGuard g, SantizedGuardTaintTrackingConfiguration taintTracker |
+        g.getEnclosingCallable() = this and
+        taintTracker
+            .hasFlow(DataFlow::parameterNode(paramFilename),
+              DataFlow::exprNode(g.getFilePathArgument())) and
+        (
+          exists(AbstractValues::BooleanValue bv |
+            // If there exists a control block that guards against misuse
+            bv.getValue() = true and
+            g.controlsNode(ret.getAControlFlowNode(), bv)
+          )
+          or
+          // Or if the function returns the resultant of the guard call
+          DataFlow::localFlow(DataFlow::exprNode(g), DataFlow::exprNode(ret.getExpr()))
+        )
+      )
+    )
   }
 }
 
-/** An access to the `FullName` property of a `ZipArchiveEntry`. */
-class ArchiveFullNameSource extends Source {
-  ArchiveFullNameSource() {
-    exists(PropertyAccess pa | this.asExpr() = pa |
+class IndirectOverloadedWrapperSantizierMethod extends AbstractWrapperSanitizerMethod {
+  /**
+   * An IndirectOverloadedWrapperSanitizerMethod is a Method in which simply wraps /another/ wrapper.class
+   *
+   * Usually this will look like the following stanza:
+   * boolean someWrapper(string s){
+   *  return someWrapper(s, true);
+   * }
+   */
+  IndirectOverloadedWrapperSantizierMethod() {
+    forex(ReturnStmt ret | ret.getEnclosingCallable() = this |
+      ret.getExpr().(BoolLiteral).getBoolValue() = false
+      or
+      exists(ZipSlipGuard g |
+        // If the parameter flows directly to SanitizerMethodCall, and the resultant is returned
+        DataFlow::localFlow(DataFlow::parameterNode(paramFilename),
+          DataFlow::exprNode(g.getFilePathArgument())) and
+        DataFlow::localFlow(DataFlow::exprNode(g), DataFlow::exprNode(ret.getExpr()))
+      )
+    )
+  }
+}
+
+class WrapperSanitizerMethodCall extends SanitizerMethodCall {
+  // A Wrapped Sanitizer Method call (some function that is equally or more restrictive than our root sanitizer)
+  AbstractWrapperSanitizerMethod wrapperMethod;
```

Comment:
I am not very well versed in this myself, but I know Ben has brought up some concern with CodeQL fields in general